objects/src/transaction/inputs.rs

```diff
@@ -402,7 +402,10 @@ fn validate_is_in_block(
         .note_path()
         .verify(note_index, note_hash, &block_header.note_root())
         .map_err(|_| {
-            TransactionInputError::InputNoteNotInBlock(note.id(), proof.location().block_num())
+            TransactionInputError::InputNoteNotInBlock(
+                note.id(),
+                proof.location().block_num().as_u32(),
+            )
```

Comment:
Should we update the error here to take `BlockNumber` (rather than a `u32`) as the second parameter?